### PR TITLE
API-2657: R4PatientTransformer supports managingOrganization field

### PR DESCRIPTION
**Pull request by jjabando1**

### CODE REVIEW

**Review by jhulbert-va — CHANGES_REQUESTED** [review]
Test?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java [hunk 1]
```diff
@@ -453,6 +452,14 @@ private String deceasedDateTime() {
     return emptyToNull(results);
   }
 
+  private Reference managingOrganization() {
+    Optional<String> managingOrganization = datamart.managingOrganization();
+    if (managingOrganization == null) {
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:457]: `managingOrganization` is an `Optional<String>` there is a better way to check this.

Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:457]: null or empty

Comment (jhulbert-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:457]: Where does `Datamart.create()` and `R4.create()` come from? Are they populating the managingOrganziation field?

Comment (jjabando1) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:457]: Ah ok those are taken from PatientSamples. I think I need to update
public gov.va.api.health.r4.api.resources.Patient patient()'s returned Patient builder to include managingOrganization

Comment (jjabando1) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:457]: So I assumed the better way to check was:
return managingOrganization.isPresent()
        ? Reference.builder().reference(managingOrganization.get()).build()
        : null;

This change causes most of the Controller and Transformer unit tests to fail. NullPointerExceptions for calls involving R4PatientTransformer.builder() and the controller(). I have no idea what is causing them to go null.

Comment (jblefkove-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:457]: `Transformers.isBlank()` is the best way to verify that an Optional is both non-null and non-empty.

However, the fact that `datamart.managingOrganization()` is returning a null reference is also troubling. The different `DatamartXXX` model classes are supposed to include lazily-initializing getters for their `Optional` fields, e.g. https://github.com/department-of-veterans-affairs/health-apis-data-query/blob/master/data-query/src/main/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/DatamartAllergyIntolerance.java#L46

The fact that `DatamartPatient` is not doing this for `managingOrganization` is an oversight and should be corrected.

Comment (bschofield-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:457]: Your reference is incomplete.

You would not be first person to ever want to convert a DatamartReference into an R4 Reference. This method can be exactly one line about 41 characters long.

Here's a hint
```
$ git grep DatamartReference data-query/src/main/java/ | grep R4
```

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java [hunk 2]
```diff
@@ -453,6 +452,14 @@ private String deceasedDateTime() {
     return emptyToNull(results);
   }
 
+  private Reference managingOrganization() {
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:455]: The big problem... you're adding production code, but you're not adding tests.

**Review by bschofield-va — DISMISSED** [review]
how about tests?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java (hunk 1 above)
Comment (jjabando1) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:457]: Should I go ahead and provide those lazy getters for the Optional fields in DatamartPatient?

Comment (jjabando1) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:457]: In DatamartPatient, managingOrganization is of type Optional<String>, so I am not sure where I should be doing any kind of DatamartReference to R4Reference conversion. In creating my Reference object, should I just assign Reference's display field to the managingOrganization string, rather than the reference field?

Comment (jblefkove-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:457]: Yes, add the lazy getters

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java (hunk 2 above)
Comment (bschofield-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:458]: You haven't actually added any tests specifically for this.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java (hunk 1 above)
Comment (bschofield-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java]: My point about the the 41 character line to implement this method has been ignored.
See https://github.com/department-of-veterans-affairs/health-apis-data-query/blob/master/data-query/src/main/java/gov/va/api/health/dataquery/service/controller/R4Transformers.java#L66

```
   return asReference(datamart.managingOrganization());
```

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java [hunk 3]
```diff
@@ -453,6 +455,20 @@ private String deceasedDateTime() {
...
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:458]: Changes this to be packaged protected and add explicit tests to verify the the couple of possible cases: null, partially populated, fully populated.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java [hunk 4]
```diff
@@ -539,6 +554,7 @@ Patient toFhir() {
         .deceasedDateTime(deceasedDateTime())
         .maritalStatus(maritalStatus())
         .contact(contacts())
+        .managingOrganization(managingOrganization(datamart.managingOrganization()))
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/patient/R4PatientTransformer.java:557]: It's more consistent for this method to remain non-static and access `datamart.managingOrganization()` internally. (In your unit test, you can set the `datamart` field on the transformer with the full test patient object and override just the `managingOrganization` field like so: `Datamart.create().patient().managingOrganization(...)`)

**Review by jblefkove-va — APPROVED** [review]

**Review by bschofield-va — CHANGES_REQUESTED** [review]
hol up

**Review by bschofield-va — APPROVED** [review]
